File: api/app/controllers/sessions_controller.ts
import type { HttpContext } from '@adonisjs/core/http'
import User from '#models/user'


export default class SessionsController {
  async store({
    auth,
    request,
    response,
  }: HttpContext) {
    const { email, password } = request.only(['email', 'password'])
    const user = await User.verifyCredentials(email, password)
    const token = await User.accessTokens.create(user)

    return {
      user: user.serialize(),
      token: {
        type: 'bearer',
        value: token.value!.release(),
      }
    }
  }
}

File: api/app/controllers/sessions_controller.rb
class SessionsController < ApplicationController

  def create
    username = params[:email]
    password = params[:password]

    user = User.authenticate_by(
      email: username,
      password: password
    )

    if user
      token = JwtService.new.create(user)
      render json: { 
        token: token,
        user: {
          id: user.id,
          email: user.email,
          firstName: user.first_name,
          lastName: user.last_name,
        }
      }
    else
      render json: { error: 'Unauthorized' }, status: :unauthorized
    end
  end

end